src/artifactContentRenderers/renderingArtifactContentRenderer.ts

```diff
@@ -0,0 +1,55 @@
+/*!
+ * Copyright (c) Microsoft Corporation. All Rights Reserved.
+ */
+
+import * as vscode from 'vscode';
+import * as sarif from 'sarif';
+import { ArtifactContentRenderer } from './artifactContentRendering';
+
+/**
+ * Class used to render artifact content objects with text.
+ */
+export class RenderingArtifactContentRenderer implements ArtifactContentRenderer {
+    /**
+     * Creates an instance of the binary content renderer.
+     * @param content The string contents to be rendered as markdown.
+     */
+    private constructor(protected readonly artifactContent: sarif.ArtifactContent) {
+        if (!artifactContent.rendered || (!artifactContent.rendered.markdown && !artifactContent.rendered.text)) {
+            throw new Error('Expected to have markdown or text content string');
+        }
+    }
+
+    /**
+     * Attempts to create an instance of the binary content renderer based on a SARIF log, run index and artifact index
+     * Returns undefined if the artifact contents cannot be found, or the content is not binary content.
+     * @param log The SARIF log.
+     * @param runIndex The run index.
+     * @param artifactIndex The artifact index.
+     */
+    public static tryCreateFromLog(log: sarif.Log, artifactContents: sarif.ArtifactContent, runIndex: number, artifactIndex: number): ArtifactContentRenderer | undefined {
+        if (!artifactContents.rendered || (!artifactContents.rendered.markdown && !artifactContents.rendered.text)) {
```

Comment:
What are your thoughts about being more DRY in general? Such as the conditions here being repeated here and Line 19.

Either check once, or have the other be a try/catch. As it stands, it risks having the two set of conditions fall out of sync.

Reply:
Just realized it's repeated on Line 43 also.